daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -751,18 +760,23 @@ private[lf] case class PartialTransaction(
     }
 
   /** Insert the given `LeafNode` under a fresh node-id, and return it */
-  def insertLeafNode(node: LeafNode, version: TxVersion): PartialTransaction = {
+  def insertLeafNode(
+      node: LeafNode,
+      version: TxVersion,
+      optLocation: Option[Location],
+  ): PartialTransaction = {
     val _ = version
     val nid = NodeId(nextNodeIdx)
     copy(
+      actionNodeLocations = actionNodeLocations :+ optLocation,
       nextNodeIdx = nextNodeIdx + 1,
       context = context.addActionChild(nid, version),
       nodes = nodes.updated(nid, node),
     )
   }
 
   /** Unwind the transaction aborting all incomplete nodes */
-  def unwind: PartialTransaction = {
+  def unwind(): PartialTransaction = {
```

Comment:
Added parens for clarity. I find paren-less `def`s unnecessarily surprising!

Reply:
The scala convention, is "use parentheses if there is a side effect". I think it also fine to add it if the evaluation is computationally heavy. 
In this particular case, I am curious to know what did you find surprising.

Reply:
If was a line in `finish()` which I found really
```
val ptx = unwind
```
It looked like a variable renaming, but of course was a method call.

Reply:
This is pretty standard scala. I think it is better you get use to it that working around. 
`val x = y` is definitely not a variable renaming in scala. Tale for example :

```
call f(x: => X) = {
    ...
    val y = x 
    ...
    g(x)
    ...
    h(x)
    ...
}
```